Move openjph to Requires.private in OpenEXR.pc

cary-ilm opened the pull request:
It's not a part of the public interface, so it's not necessary for application linking.

Also, since `src/lib/OpenEXRCore/CMakeLists.txt` declares OpenJPH as private:

` 
target_link_libraries(OpenEXRCore PRIVATE ${EXR_OPENJPH_LIB})
`

it is *not* necessary in the `target_link_libraries(PUBLIC)` declaration in cmake/LibraryDefine.cmake, so this change removes it there.

This also adds a ci test to validate the OpenEXR.pc.

Should resolve #2111.

cary-ilm committed:
Remove openjph from OpenEXR.pc

It's not a part of the public interface, so it's not necessary for
application.

Also, since src/lib/OpenEXRCore/CMakeLists.txt declares OpenJPH as
private:

 target_link_libraries(OpenEXRCore PRIVATE ${EXR_OPENJPH_LIB})

it is *not* necessary in the `target_link_libraries(PUBLIC)`
declaration in cmake/LibraryDefine.cmake, so this change removes it
there.

This also adds a ci test to validate the OpenEXR.pc.

Signed-off-by: Cary Phillips <cary@ilm.com>

meshula reviewed (approved):
lgtm, thanks for the clarification on slack.

kmilos commented on cmake/OpenEXR.pc.in:
```diff
@@ -17,6 +17,5 @@ Version: @OPENEXR_VERSION@
 Cflags: -I${includedir} -I${OpenEXR_includedir} @exr_pthread_cflags@
 Requires: Imath
-Requires: openjph
 Requires.private: @EXR_DEFLATE_PKGCONFIG_REQUIRES@
```
If used, should still go to `Requires.private` for static linking, just like libdeflate.

cary-ilm committed:
Add EXR_OPENJPH_PKGCONFIG_REQUIRES to Requires.private

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm committed:
Add -static to validate_pkgconfig.sh for static builds

Signed-off-by: Cary Phillips <cary@ilm.com>

cary-ilm commented:
OK, I made that change, and changed the PR title to match. I also fixed the validate_pkgconfig.sh CI test to add the `-static` flag, which appears to be necessary then for static builds.

kmilos commented:
> OK, I made that change, and changed the PR title to match. I also fixed the validate_pkgconfig.sh CI test to add the `-static` flag, which appears to be necessary then for static builds.

That's great, thanks!
